### DISCUSSION TITLE
Errors in macros plugin result in zero exit code

### DISCUSSION THREAD

**Discussion by tovrstra** [mkdocs/mkdocs#3734] [Q&A] [answered]
### Steps to reproduce

Create `doc/index.md` as follows:

```markdown
# Hello

{% include 'doesnotexist.md' %}
```

Create `mkdocs.yaml` as follows:

```yaml
site_name: Hello

strict: true
validation:
  nav:
    omitted_files: warn
    absolute_links: warn
  links:
    anchors: warn
    absolute_links: warn
    unrecognized_links: warn

plugins:
- macros
```

Run mkdocs:

```bash
mkdocs build
```

The terminal output contains the expected error:

```
INFO    -  [macros] - Macros arguments: {'module_name': 'main', 'modules': [], 'render_by_default': True, 'include_dir':
           '', 'include_yaml': [], 'j2_block_start_string': '', 'j2_block_end_string': '', 'j2_variable_start_string': '',
           'j2_variable_end_string': '', 'on_undefined': 'keep', 'on_error_fail': False, 'verbose': False}
INFO    -  [macros] - Extra filters (module): ['pretty']
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: .../tmp/site
INFO    -  [macros] - ERROR # _Macro Rendering Error_

           _File_: `index.md`

           _TemplateNotFound_: doesnotexist.md

           ```
           Traceback (most recent call last):
             File ".../venv/lib64/python3.12/site-packages/mkdocs_macros/plugin.py", line 527, in
           render
               return md_template.render(**page_variables)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
             File ".../venv/lib64/python3.12/site-packages/jinja2/environment.py", line 1304, in
           render
               self.environment.handle_exception()
             File ".../venv/lib64/python3.12/site-packages/jinja2/environment.py", line 939, in
           handle_exception
               raise rewrite_traceback_stack(source=source)
             File "<template>", line 3, in top-level template code
             File ".../venv/lib64/python3.12/site-packages/jinja2/loaders.py", line 207, in
           get_source
               raise TemplateNotFound(template)
           jinja2.exceptions.TemplateNotFound: doesnotexist.md

           ```
INFO    -  Documentation built in 0.08 seconds
```

However, the exit code is still zero:

```bash
echo $?
```

gives

```
0
```

### Expected result

Non-zero exit code

**Comment by pawamoy** [comment]
Tagging @fralau and moving to Q/A :slightly_smiling_face:

**Comment by fralau** [comment, marked as answer]
You might want to update your version of Mkdocs-Macros, from github.

According to [your settings](https://mkdocs-macros-plugin.readthedocs.io/en/latest/troubleshooting/#is-it-possible-to-make-the-building-process-fail-in-case-of-page-error), you will get a warning or an error.

**Reply by tovrstra** [reply]
Thanks. I can confirm that, after installation of the plugin from the master branch, the exitcode is indeed non-zero for my example.

In case someone wants to copy-paste the commands:

```bash
pip install git+https://github.com/fralau/mkdocs-macros-plugin@9a08f11742c130b06ff3345d64855669dbbcc4aa
mkdocs build
```

(You can leave out the `@9a08f11742c130b06ff3345d64855669dbbcc4aa` part if you also want to get changes from after the time of writing.)
